app/uk/gov/hmrc/tariffclassificationfrontend/controllers/CaseController.scala

```diff
@@ -72,6 +72,20 @@ class CaseController @Inject()(authenticatedAction: AuthenticatedAction,
     })
   }
 
+  def attachmentsDetails(reference: String): Action[AnyContent] = authenticatedAction.async { implicit request =>
+    getCaseAndRenderView(reference, CaseDetailPage.ATTACHMENTS, c => {
+      val view = for {
+        attachments <- fileStoreService.getAttachments(c)
+        letter <- fileStoreService.getLetterOfAuthority(c)
+      } yield {
+        val filesFromApplicant = attachments.toStream.filter(_.application)
```

Comment:
this code can be improved

Reply:
Try to split the original list into 2 pieces using `partition`
https://stackoverflow.com/questions/12148581/how-to-split-a-sequence-into-two-pieces-by-predicate